File: src/rngGame/stats/Item.java
package rngGame.stats;

import javafx.scene.image.Image;
import rngGame.tile.ImgUtil;

public class Item {

	public static final Item NO_ITEM = new Item("Use", "noItem", Rarity.COMMON);

	private Rarity rarity;

	private String path;

	public Item(String ordner, String item, Rarity rarity) {
		this.rarity = rarity;

		path = "./res/Items/" + ordner + "/" + item + ".png";
	}

	protected void setRarity(Rarity rarity) {
		this.rarity = rarity;
	}

	protected void setPath(String path) {
		this.path = path;
	}

	public Rarity getRarity() {
		return rarity;
	}

	public String getPath() {
		return path;
	}

	public Image getImage() {
		return ImgUtil.getScaledImage(path);
	}

}

File: src/rngGame/stats/Item.kt
package rngGame.stats

import javafx.scene.image.Image
import rngGame.tile.ImgUtil

open class Item(ordner: String, item: String, rarity: Rarity) {

	var rarity: Rarity = rarity
		protected set

	var path = "./res/Items/$ordner/$item.png"
		protected set

	val image: Image
		get() = ImgUtil.getScaledImage(path)

	override fun toString(): String {
		return "Item [path=$path, rarity=$rarity]"
	}

	companion object {
		val NOITEM = Item("Use", "noItem", Rarity.COMMON)
	}

}
